tools/changelog/changelog/validation.go

```diff
@@ -0,0 +1,85 @@
+package changelog
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ValidateType ensures a type string is in the permitted list and always returns a non-empty string if given a
+// non-empty string to allow "--force" behavior.
+func ValidateType(config Config, typ string) (string, error) {
+	typ = strings.ToLower(strings.TrimSpace(typ))
+	for _, v := range config.Types.Keys() {
+		if typ == v {
+			return typ, nil
+		}
+	}
+
+	return typ, fmt.Errorf("unknown entry type %q", typ)
+}
+
+func PermittedTypesString(config Config) string {
+	return strings.Join(config.Types.Keys(), ", ")
+}
+
+// var languages []string = []string{
```

Comment:
Leftovers? Is this stuff all config driven now?

Reply:
Yeah, will do a comment sweep before making PR to pulumictl.